code/modules/reagents/chemistry/holder.dm

```diff
@@ -476,6 +504,25 @@
 		del_reagent(R.id)
 	return 0
 
+/datum/reagents/proc/reaction_check(mob/living/M, datum/reagent/R)
+	var/can_process = FALSE
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		//Check if this mob's species is set and can process this type of reagent
+		if(H.dna && H.dna.species.reagent_tag)
+			if((R.process_flags & SYNTHETIC) && (H.dna.species.reagent_tag & PROCESS_SYNTHETIC))		//SYNTHETIC-oriented reagents require PROCESS_SYNTHETIC
+				can_process = TRUE
+			if((R.process_flags & ORGANIC) && (H.dna.species.reagent_tag & PROCESS_ORGANIC))		//ORGANIC-oriented reagents require PROCESS_ORGANIC
+				can_process = TRUE
+			//Species with PROCESS_DUO are only affected by reagents that affect both organics and synthetics, like acid and hellwater
+			if((R.process_flags & ORGANIC) && (R.process_flags & SYNTHETIC) && (H.dna.species.reagent_tag & PROCESS_DUO))
```

Comment:
Should be:
`if((R.process_flags & (ORGANIC | SYNTHETIC)) && (H.dna.species.reagent_tag & (PROCESS_ORGANIC | PROCESS_SYNTHETIC))`

For reagents that work on both Synths and Organics. No species should have both flags enabled, as far as I can tell.

Also, I'd cut down the flags into just one set - `PROCESS_ORGANIC` and `PROCESS_SYNTHETIC`